clusterctl/clusterdeployer/clusterdeployer.go

```diff
@@ -411,92 +437,94 @@ func pivot(from, to ClusterClient) error {
 	}
 
-	fromMachineSets, err := from.GetMachineSetObjects()
+	fromMachineSets, err := from.GetMachineSetObjectsInNamespace(namespace)
 	if err != nil {
 		return err
 	}
 	for _, machineSet := range fromMachineSets {
 		// New objects cannot have a specified resource version. Clear it out.
 		machineSet.SetResourceVersion("")
-		if err := to.CreateMachineSetObjects([]*clusterv1.MachineSet{machineSet}); err != nil {
+		if err := to.CreateMachineSetObjects([]*clusterv1.MachineSet{machineSet}, namespace); err != nil {
 			return fmt.Errorf("error moving MachineSet '%v': %v", machineSet.GetName(), err)
 		}
 		glog.Infof("Moved MachineSet %v", machineSet.GetName())
 	}
 
-	machines, err := from.GetMachineObjects()
+	machines, err := from.GetMachineObjectsInNamespace(namespace)
 	if err != nil {
 		return err
 	}
 
 	for _, machine := range machines {
 		// New objects cannot have a specified resource version. Clear it out.
 		machine.SetResourceVersion("")
-		if err = to.CreateMachineObjects([]*clusterv1.Machine{machine}); err != nil {
+		if err = to.CreateMachineObjects([]*clusterv1.Machine{machine}, namespace); err != nil {
 			return fmt.Errorf("error moving Machine '%v': %v", machine.GetName(), err)
 		}
 		glog.Infof("Moved Machine '%s'", machine.GetName())
 	}
 	return nil
 }
 
-func deleteObjects(client ClusterClient) error {
+func deleteObjectsInNamespace(client ClusterClient, namespace string) error {
 	var errors []string
-	glog.Infof("Deleting machine deployments")
-	if err := client.DeleteMachineDeploymentObjects(); err != nil {
+	glog.Infof("Deleting machine deployments in namespace %q", namespace)
+	if err := client.DeleteMachineDeploymentObjectsInNamespace(namespace); err != nil {
 		err = fmt.Errorf("error deleting machine deployments: %v", err)
 		errors = append(errors, err.Error())
 	}
-	glog.Infof("Deleting machine sets")
-	if err := client.DeleteMachineSetObjects(); err != nil {
+	glog.Infof("Deleting machine sets in namespace %q", namespace)
+	if err := client.DeleteMachineSetObjectsInNamespace(namespace); err != nil {
 		err = fmt.Errorf("error deleting machine sets: %v", err)
 		errors = append(errors, err.Error())
 	}
-	glog.Infof("Deleting machines")
-	if err := client.DeleteMachineObjects(); err != nil {
+	glog.Infof("Deleting machines in namespace %q", namespace)
+	if err := client.DeleteMachineObjectsInNamespace(namespace); err != nil {
 		err = fmt.Errorf("error deleting machines: %v", err)
 		errors = append(errors, err.Error())
 	}
-	glog.Infof("Deleting clusters")
-	if err := client.DeleteClusterObjects(); err != nil {
+	glog.Infof("Deleting clusters in namespace %q", namespace)
+	if err := client.DeleteClusterObjectsInNamespace(namespace); err != nil {
 		err = fmt.Errorf("error deleting clusters: %v", err)
 		errors = append(errors, err.Error())
 	}
+	glog.Infof("Deleting namespace %q", namespace)
+	if err := client.DeleteNamespace(namespace); err != nil {
```

Comment:
I believe that this will always fail for the default namespace. Have you tested this flow both with a new namespace and the default namespace?